### PR TITLE
Enhancement/issues#1 initial configuration

### PR DESCRIPTION
**Pull request by cxoltero**
resoves #1 

@thescientist13 | @paul-barry-kenzan | @thisgeek | @ChrisPlease

### CODE REVIEW

**Comment by thescientist13** [comment]
ok to test

**Comment by thescientist13** [comment]
Can one of the admins verify this patch?

**Comment by thescientist13** [comment]
@cxoltero 
Please add a build script and then this can get merged
```
09:35:32 [CI-angular-seed] $ /bin/sh -xe /tmp/hudson3294584095974593338.sh
09:35:32 + sh bin/build.sh
09:35:32 sh: 0: Can't open bin/build.sh
09:35:32 Build step 'Execute shell' marked build as failure
```

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["2015"]
```
Comment (thescientist13) [inline .babelrc]: let's match react seed
https://github.com/kenzanlabs/react-seed/blob/master/.babelrc

File: README.md [hunk 2]
```diff
@@ -1,2 +1,109 @@
...
+
+### Application Files
+
+ Application code, including unit tests. Directories are intended to be
+ kept as flat as possible with a B.O.F. (birds of a feather) organization.
+
+```
+angular-seed
+│
+└───src
```
Comment (thescientist13) [inline README.md]: this isn't quite accurate given our layout in react-seed, though it seems that it should be updated there as well too. 

It should reflect
- src/views
- src/services
- src/components
- etc

File: webpack.config.common.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  context: path.resolve(__dirname, 'src'),
+
+  // Enable sourcemaps for debugging webpack's output.
+  devtool: 'source-map',
+
+  resolve: {
+    extensions: ['.ts', '.js']
+  },
+
+  module: {
```
Comment (thescientist13) [inline webpack.config.common.js]: let's make sure this matches what we have in react-seed
https://github.com/kenzanlabs/react-seed/blob/master/webpack.config.common.js

**Comment by paul-barry-kenzan** [comment]
@cxoltero @thescientist13 
Hasn't some (maybe most) of this work been done already in the React Seed? Isn't this second repo the perfect use case for the seed? Why not use the seed as it's intended to start repos?  Clone, strip and push is all that we should need, right? Or as I suggested previously, maintain a slimmed down repo with this initial config already in place ready for the Framework implementation, eliminating the need to strip out anything.

**Comment by thescientist13** [comment]
@paul-barry-kenzan 
I suppose, yes we could have cloned from an early hash from the other repo, but in the end, no one would have probably noticed.

But IMO, our goal here is not achieve architectural perfection per se.  The intent was first to provide a learning exercise for contributors (like @cxoltero ) and the readers of our related article series.  Keeping things simple was by far the main objective over anything else.  

To that point, maybe _seed_ was a bad name for these repos?

**Comment by paul-barry-kenzan** [comment]
@thescientist13 
Makes sense. When I saw your comments about "let's match react seed", I was curious as to why we weren't following the recommended workflow of having the seeds as a starting point.

As for the names, maybe the 2 current repos become "POC", while a 3rd "seed" is created? Just a thought?

**Comment by thescientist13** [comment]
@paul-barry-kenzan 
I was thinking of just something like **react-webpack-stack**?

> "let's match react seed"
Yeah, only because I want as much apples to apples as possible in terms of everything not specific to the Angular or React parts of the stack (eg. differing npm packages, karma vs jest, as examples).

I certainly think there is a case to what you are proposing.  I have documented something similar for my personal projects; [Evergreen Build](https://github.com/thegreenhouseio/php-api-seed/wiki/Starting-a-new-Project#cloning-evergreen-build), I call it).   We could support an Evergreen "vanilla-webpack-stack" for ESLint, Babel, and maybe webpack configs?

That said, I would like to see how these two repos stack against each other (pun strongly intended) before we go down that route, to make sure we have a clear vision on what can / should be shared.

**Comment by thescientist13** [comment]
ok to test

**Comment by thescientist13** [comment]
ok to test

**Comment by thescientist13** [comment]
Can one of the admins verify this patch?

**Comment by cxoltero** [comment]
Hot module replacement is configured as library agnostic at this point. Once Angular is added it will require a different configuration.

**Comment by thescientist13** [comment]
Thanks @cxoltero!

I finally got a semblence of CircleCI working here
https://github.com/kenzanlabs/react-webpack-stack/pull/51/

Just a couple more TODO's then I will set it up for this repo as well.

**Comment by thescientist13** [comment]
see #7

**Pull request closed without merging**